Merged bundle support

scf2k opened the pull request.

arikon committed:
Fix unresolved promise in BemBuildNode.isValid()

In case when BEM_IO_STRATEGY === 'callback' and `meta` was empty promise would never resolve.
Add `meta` size checking to fix this.

arikon committed:
Use bem-bl as git submodule for tests data: remove bem-bl

See bem/bem-tools#176

arikon committed:
Use bem-bl as git submodule for tests data: use bem-bl as submodule, update reference test results

See bem/bem-tools#176

Commit:
Wrong link to ometa-js. Fixed

Commit:
Wrong link to ometa-js. Fixed in russian doc too

Commit:
- merged bundle support is added

Commit:
Merge branch 'server' of github.com:bem/bem-tools into merged-bundle

arikon commented on lib/nodes/mergedbundle.js:
```diff
@@ -0,0 +1,79 @@
+        return node;
+    }
+
+});
```
Контент этого файла можно положить в `nodes/bundle.js`, чтобы всё про бандл было в одном месте

arikon commented on lib/nodes/magic.js:
```diff
@@ -1,8 +1,15 @@
+    MagicNodeName = exports.MagicNodeName = 'MagicNode';
+
+exports.__defineGetter__(MagicNodeName, function() {
+return registry.getNodeClass(MagicNodeName);
```
Отступа не хватает

arikon commented on test/nodes-lib.js:
```diff
@@ -3,7 +3,8 @@ var assert = require('chai').assert,
     QFS = require('q-fs'),
 
-    SymlinkLibraryNode = require('../lib/nodes').SymlinkLibraryNode;
+    SymlinkLibraryNode = require('../lib/nodes/lib');
```
`SymlinkLibraryNode` -> `LibNodes`, в остальных местах по аналогии

arikon commented on lib/nodesregistry.js:
```diff
@@ -0,0 +1,68 @@
+    // (nodeName, baseName, object)
+    // (nodeName, baseName, object, static)
+
+    _decl: function(nodeName, objectOrBaseName, objectOrStatic, staticObject) {
```
Напиши честный jsdoc

arikon commented on lib/nodesregistry.js:
```diff
@@ -0,0 +1,68 @@
+    registry = {},
+    cache = {};
+
+var Make = exports.NodesRegistry = {
```
`module.exports = {...}`

Здесь же один объект на весь модуль.

arikon commented on lib/nodes/lib.js:
```diff
@@ -6,12 +6,19 @@ var Q = require('q'),
-    Node = require('./node').Node,
+    Node = require('./node').NodeName,
+
+    LibraryNodeName = exports.LibraryNodeName = 'LibraryNode',
```
Я так и не понял, для чего хранить имена классов в переменных

arikon commented:
/cc @veged

arikon commented on test/nodes-lib.js:
```diff
@@ -20,7 +21,7 @@ describe('nodes', function() {
     describe('new SymlinkLibraryNode()', function() {
 
-        var node = new SymlinkLibraryNode({
+        var node = new SymlinkLibraryNode.SymlinkLibraryNode({
```
`SymlinkLibraryNode.SymlinkLibraryNode` — это упячечно =(

scf2k replied:
Ну если бы не было несколько классов в одном файле, можно было бы сделать не упячечно

arikon replied:
Я выше предложил, как сделать неупячечно

Commit:
- a bit of refactoring

Commit:
- added test for merged bundle

Commit:
- a little of jsdoc

Commit:
- refactored test for merged bundle

Merge branch 'server' of github.com:bem/bem-tools into merged-bundle

Conflicts:
	test/nodes-lib.js

Commit:
- removed unused var

Commit:
- removed unused code/files

arikon committed:
Code style fixes
